e2e/scenario/scenario_ubuntu2204-artifact-streaming.go

```diff
@@ -0,0 +1,31 @@
+package scenario
+
+import (
+	"github.com/Azure/agentbaker/pkg/agent/datamodel"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+)
+
+func ubuntu2204ArtifactStreaming() *Scenario {
+	return &Scenario{
+		Name:        "ubuntu2204-artifact-streaming",
+		Description: "tests that a new ubuntu 2204 node using artifact streaming can be properly bootstrapepd",
+		Config: Config{
+			ClusterSelector: NetworkPluginKubenetSelector,
+			ClusterMutator:  NetworkPluginKubenetMutator,
+			BootstrapConfigMutator: func(nbc *datamodel.NodeBootstrappingConfiguration) {
+				nbc.EnableArtifactStreaming = true
+				nbc.ContainerService.Properties.AgentPoolProfiles[0].Distro = "aks-ubuntu-containerd-22.04-gen2"
+				nbc.AgentPoolProfile.Distro = "aks-ubuntu-containerd-22.04-gen2"
+			},
+			VMConfigMutator: func(vmss *armcompute.VirtualMachineScaleSet) {
+				vmss.Properties.VirtualMachineProfile.StorageProfile.ImageReference = &armcompute.ImageReference{
+					ID: to.Ptr(DefaultImageVersionIDs["ubuntu2204"]),
+				}
+			},
+			LiveVMValidators: []*LiveVMValidator{
+				NonEmptyDirectoryValidator("/etc/overlaybd"),
```

Comment:
Is checking that this dir is not empty the only "proof" we need to be sure that artifact streaming is properly bootstrapped?

Reply:
It's necessary but not sufficient - we need to check other systemd units are enabled too (like overlaybd-snapshotter, overlaybd-tcmu). I'm planning to add a custom `ArtifactStreamingValidator` that returns a `LiveVMValidator` in a future PR.